types/superagent-retry-delay/index.d.ts

```diff
@@ -0,0 +1,72 @@
+// Type definitions for superagent-retry-delay 2.6
+// Project: https://github.com/luispabon/superagent-retry-delay#readme
+// Definitions by: Aaron Ross <https://github.com/me>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.9
+
+import { SuperAgentRequest, SuperAgentStatic, Request, Response } from "superagent";
+import * as stream from "stream";
+import * as cookiejar from "cookiejar";
+
+// these types should be exported by `superagent` but are not -------
+
+type CallbackHandler = (err: any, res: Response) => void;
+
+type Serializer = (obj: any) => string;
+
+type BrowserParser = (str: string) => any;
+
+type NodeParser = (res: Response, callback: (err: Error | null, body: any) => void) => void;
+
+type Parser = BrowserParser | NodeParser;
+
+// ------------------------------------------------------------------
+
+type EnhancedSuperAgentRequest = Omit<SuperAgentRequest, "retry"> & {
+    retry(count?: number, delay?: number): SuperAgentRequest;
+};
+
+interface EnhancedSuperAgent<Req extends EnhancedSuperAgentRequest> extends stream.Stream {
```

Comment:
I'd love to be able to just extend `SuperAgent` directly and override the generic constraint for `Req`, but I can't seem to find a way to express that correctly; suggestions welcome here (same goes for `EnhancedSuperAgentStatic`).